src/main/java/org/folio/sample/service/BookService.java

```diff
@@ -32,10 +32,18 @@ public Optional<Book> getBookById(UUID id) {
   public Book createBook(Book book) {
     return bookRepository.save(book);
   }
-  public Book updateBook(UUID id, String name, LocalDate publishedDate) {
-    Book curBook = bookRepository.getReferenceById(id);
-    curBook.setName(name);
-    curBook.setPublishedDate(publishedDate);
-    return bookRepository.save(curBook);
+
+  public Optional<Book> updateBook(UUID bookId, Book book) {
+    // Get the book from the database, throw NotFoundException if not found
+    Book bookFromDb = bookRepository.findById(bookId).orElseThrow(() -> new NotFoundException("Book not found"));
+
+    // Update the book
+    bookFromDb.setName(book.getName());
+    bookFromDb.setPublishedDate(book.getPublishedDate());
+
+    
+    // Save the updated book
+    return Optional.of(bookRepository.save(bookFromDb));
```

Comment:
Any reason we're returning an `Optional` here?  It looks like `.save` [will never be null](https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html#save(S))

Reply:
It was giving me errors when I tried to just return the Book, because my return type is Optional<Book>.

Reply:
But I guess the return type can just be Book because the orElseThrows handles the case in which we have an invalid id, right?

Reply:
You can make your return type just `Book`, since it'll always be there

Reply:
Yep, exactly. When you throw an exception, it leaves the normal return path, so we can assume that "if the function returns, it found the book". Optional provides a wrapper around nulls/nonexistent values, protects use from `NullPointerException`s (Java's segfault) and also ensures we handle non present values correctly (instead of accidentally forgetting an `if (whatever == null)`)